File: Examples/VisualBasic/WorkingWithProjects/Miscellaneous/ExtractEmbeddedObjects.vb
Imports System.IO
Imports Aspose.Tasks
'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Tasks for .NET API reference 
'when the project is build. Please check https://docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Tasks for .NET API from http://www.aspose.com/downloads, 
'install it and then add its reference to this project. For any issues, questions or suggestions 
'please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace WorkingWithProjects.Miscellaneous
    Public Class ExtractEmbeddedObjects
        Public Shared Sub Run()
            ' ExStart:ExtractEmbeddedObjects
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            Dim project As New Project(dataDir & Convert.ToString("Project1.mpp"))
            Dim ole As OleObject = project.OleObjects.ToList()(0)

            ' We have to check this property because it can be null if the embedded object was created inside the ms project application Or, alternatively, you can use this check: if (ole.FileFormat == "Package")
            If Not String.IsNullOrEmpty(ole.FullPath) Then
                Using fileStream As New FileStream(dataDir, FileMode.Create)
                    fileStream.Write(ole.Content, 0, ole.Content.Length)
                End Using
            End If
            ' ExEnd:ExtractEmbeddedObjects
        End Sub
    End Class
End Namespace

File: Examples/CSharp/WorkingWithProjects/Miscellaneous/ExtractEmbeddedObjects.cs
namespace Aspose.Tasks.Examples.CSharp.WorkingWithProjects.Miscellaneous
{
    using System.IO;

    internal class ExtractEmbeddedObjects
    {
        public static void Run()
        {
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod());

            //ExStart:ExtractEmbeddedObjects
            //ExFor: OleObject.Content
            //ExFor: Project.OleObjects
            //ExSummary: Shows how to extract an embedded OLE object.
            var project = new Project(dataDir + "ExtractEmbeddedObjects.mpp");

            foreach (var ole in project.OleObjects.ToList())
            {
                // We have to check this property because it can be null if the embedded object was created inside the ms project application Or, alternatively, you can use this check: if (ole.FileFormat == "Package")
                if (string.IsNullOrEmpty(ole.FullPath) || ole.Content == null)
                {
                    continue;
                }

                using (var stream = new FileStream(dataDir + Path.GetFileName(ole.FullPath), FileMode.Create))
                {
                    stream.Write(ole.Content, 0, ole.Content.Length);
                }
            }
            //ExEnd:ExtractEmbeddedObjects
        }
    }
}